src/streaming-ingestion/flink-etl/src/main/java/software/aws/solution/clickstream/flink/StreamingJob.java

```diff
@@ -0,0 +1,419 @@
+ import org.apache.flink.api.common.functions.FilterFunction;
+ import org.apache.flink.api.common.serialization.SimpleStringSchema;
+ import org.apache.flink.api.common.typeinfo.Types;
+ import org.apache.flink.api.java.tuple.Tuple2;
+ import org.apache.flink.api.java.utils.ParameterTool;
+ import org.apache.flink.connector.aws.config.AWSConfigConstants;
+ import org.apache.flink.connector.kinesis.sink.KinesisStreamsSink;
+ import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.JsonNode;
+ import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.ObjectMapper;
+ import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.node.ArrayNode;
+ import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.node.ObjectNode;
+ import org.apache.flink.streaming.api.datastream.DataStream;
+ import org.apache.flink.streaming.api.datastream.SingleOutputStreamOperator;
+ import org.apache.flink.streaming.api.environment.LocalStreamEnvironment;
+ import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+ import org.apache.flink.streaming.connectors.kinesis.FlinkKinesisConsumer;
+ import org.apache.flink.streaming.connectors.kinesis.config.ConsumerConfigConstants;
+ import org.apache.flink.util.Collector;
+ import org.slf4j.Logger;
+ import org.slf4j.LoggerFactory;
+ import software.aws.solution.clickstream.plugin.enrich.Enrichment;
+ import software.aws.solution.clickstream.plugin.enrich.IPEnrichment;
+ import software.aws.solution.clickstream.plugin.transformer.DeviceTransformer;
+ import software.aws.solution.clickstream.plugin.transformer.KvTransformer;
+ import software.aws.solution.clickstream.plugin.transformer.Transformer;
+ import software.aws.solution.clickstream.plugin.transformer.URITransformer;
+
+ import java.io.*;
+ import java.nio.charset.StandardCharsets;
+ import java.time.Instant;
+ import java.time.LocalDateTime;
+ import java.time.ZoneId;
+ import java.time.format.DateTimeFormatter;
+ import java.util.*;
+ import java.util.zip.GZIPInputStream;
+
+ import static org.apache.flink.streaming.connectors.kinesis.config.ConsumerConfigConstants.*;
+ import static org.apache.flink.streaming.connectors.kinesis.config.ConsumerConfigConstants.RecordPublisherType.EFO;
+
+
+/**
+ * {
+ *     "appId": "app1",
+ *     "compression": "gzip",
+ *     "data": "",
+ *     "date": "2023-11-08T12:33:33+00:00",
+ *     "fakeIp": "1.26.162.45",
+ *     "ingest_time": 1699446813000,
+ *     "ip": "3.141.3.4",
+ *     "method": "POST",
+ *     "path": "/collect",
+ *     "platform": "Android",
+ *     "rid": "2b8613c5569f48fb9f569f2531403580",
+ *     "server_ingest_time": 1699446813000,
+ *     "source_type": "http_server",
+ *     "timestamp": "2023-11-08T12:33:33.282166261Z",
+ *     "ua": "python-requests/2.25.1",
+ *     "uri": "/collect?platform=Android&appId=app1&compression=gzip&fakeIp=1.26.162.45&event_bundle_sequence_id=100"
```

Comment:
Is it an sample? Then replace `faceIp` with `ip`

Reply:
`fakeIp` for Clickstream ingestion server debug.